src/client_side_reply.cc

```diff
@@ -1062,7 +1062,7 @@ clientReplyContext::storeNotOKTransferDone() const
     assert(mem != nullptr);
     assert(http->request != nullptr);
 
-    if (mem->baseReply().pstate != Http::Message::psParsed)
+    if (!http->storeEntry()->hasParsedReplyHeader())
```

Comment:
Please consider moving the `mem` object creation after this now that it is no longer needed beforehand. I do not insist on this.

Reply:
Sorry, this PR was merged before I saw your change request.